packages/visx-shape/src/util/getSplitLineSegments.ts

```diff
@@ -1,80 +1,110 @@
+
+export interface GetLineSegmentsConfig {
   /** Full path `d` attribute to be broken up into `n` segments. */
   path: string;
+  /** Array of length `n`, where `n` is the number of segments. */
+  pointsInSegments: PointInSegment[][];
+  /**
+   * How to segment the line
+   * - `x`: Split based on x-position,
+   *  assuming x values increase only (`segment[i].x > segment[i-1].x`)
+   *  or decrease only (`segment[i].x < segment[i-1].x`).
+   * - `y`: Split based on y-position,
+   *  assuming y values increase only (`segment[i].y > segment[i-1].y`)
+   *  or decrease only (`segment[i].y < segment[i-1].y`).
+   * - `length`: Assuming the path length between consecutive points are equal.
+   *
+   * Default is `x`.
+   */
+  segmentation: LineSegmentation;
   /**
-   * Array of length `n`, where `n` is the number of resulting line segments.
-   * For each segment of length `m`, `m / sampleRate` evenly spaced points will be returned.
+   * The `path` will be sampled every `sampleRate` pixel to generate the returned points.
+   * Default is `1` pixel.
    */
-  segments: Datum[][];
-  /** For each segment of length `m`, `m / sampleRate` evenly spaced points will be returned. */
   sampleRate?: number;
 }
 
-type LineSegments = { x: number; y: number }[][];
-
-export function getSplitLineSegments<Datum>({
+export default function getSplitLineSegments({
   path,
-  segments,
-  sampleRate = 0.25,
-}: GetLineSegmentsConfig<Datum>): LineSegments {
+  pointsInSegments,
+  segmentation = 'x',
+  sampleRate = 1,
+}: GetLineSegmentsConfig): LineSegments {
   try {
-    let pathElement = document.getElementById(MEASUREMENT_ELEMENT_ID) as SVGPathElement | null;
-
-    // create a single path element if not done already
-    if (!pathElement) {
-      const svg = document.createElementNS(SVG_NAMESPACE_URL, 'svg');
-      // not visible
-      svg.style.opacity = '0';
-      svg.style.width = '0';
-      svg.style.height = '0';
-      // off screen
-      svg.style.position = 'absolute';
-      svg.style.top = '-100%';
-      svg.style.left = '-100%';
-      // no mouse events
-      svg.style.pointerEvents = 'none';
-      pathElement = document.createElementNS(SVG_NAMESPACE_URL, 'path');
-      pathElement.setAttribute('id', MEASUREMENT_ELEMENT_ID);
-      svg.appendChild(pathElement);
-      document.body.appendChild(svg);
-    }
-
+    const pathElement = getOrCreateMeasurementElement(MEASUREMENT_ELEMENT_ID);
     pathElement.setAttribute('d', path);
+    const totalLength = pathElement.getTotalLength();
 
-    const totalPathLength = pathElement.getTotalLength();
-    const totalPieces = segments.reduce((sum, curr) => sum + curr.length, 0);
-    const pieceSize = totalPathLength / totalPieces;
+    const numSegments = pointsInSegments.length;
+    const lineSegments: LineSegments = pointsInSegments.map(() => []);
 
-    let cumulativeSize = 0;
+    if (segmentation === 'x' || segmentation === 'y') {
+      const segmentBegins = pointsInSegments.map(
```

Comment:
wdyt? starts is a little more clearly the place where each segment starts

```suggestion
      const segmentStarts = pointsInSegments.map(
```